mbedtls/src/rng/mod.rs

```diff
@@ -38,3 +38,14 @@ pub trait Random: RngCallback {
 }
 
 impl<'r, F: RngCallback> Random for F {}
+
+#[cfg(any(feature = "rdrand", target_env = "sgx"))]
+#[no_mangle]
+pub unsafe extern "C" fn mbedtls_psa_external_get_random(
```

Comment:
More specifically, this function is needed in case of non `std`. I may change to `not(feature = "std")` to better suit cases.

Since TLS 1.3 need `MBEDTLS_PSA_CRYPTO_C` , ref: mbedtls-sys/vendor/docs/architecture/tls13-support.md

But it need either `MBEDTLS_ENTROPY_C` or `MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG` defined, see: https://github.com/fortanix/rust-mbedtls/blob/0f2290561ab92a1f844b091ba1f106953ce4bc04/mbedtls-sys/vendor/include/mbedtls/check_config.h#L696-L701

So, in case without `std`, we need to provide `mbedtls_psa_external_get_random` .

Reply:
https://lists.trustedfirmware.org/archives/list/mbed-tls@lists.trustedfirmware.org/thread/NSTVGANVY6CKWIV37S4BLY6GZTZTZE3F/